include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -1174,20 +1174,28 @@ __pattern_inplace_merge(_ExecutionPolicy&& __exec, _Iterator __first, _Iterator
 //------------------------------------------------------------------------
 // sort
 //------------------------------------------------------------------------
-template <typename _ExecutionPolicy, typename _Iterator, typename _Compare>
-oneapi::dpl::__internal::__enable_if_hetero_execution_policy<_ExecutionPolicy, void>
-__pattern_sort(_ExecutionPolicy&& __exec, _Iterator __first, _Iterator __last, _Compare __comp,
-               /*vector=*/::std::true_type, /*parallel=*/::std::true_type, /*is_move_constructible=*/::std::true_type)
+template <typename _ExecutionPolicy, typename _Iterator, typename _Compare, typename _Proj>
+void
+__stable_sort_with_projection(_ExecutionPolicy&& __exec, _Iterator __first, _Iterator __last, _Compare __comp,
```

Comment:
Why is this function necessary for? is that to hide projections interface from glue_algorithm?

I am thinking that probably having projections called from glue_algorithms is a right thing to do... It might require more code changes but I am trying to understand what is more correct design-wise

Reply:
1) With design point of view it is an internal method, just for reduce code duplication.
My introduced name was `__stable_sort_impl`.
`__stable_sort_with_projection` was proposed by Alexey Kukanov.
(In my guess  - by self-documented reasons).
Once again - it is an internal method - so any readable name is appropriate here, IMHO.
2) I don't understand what you mean talking about " is more correct design-wise".

Reply:
I am trying to understand what is the right way to use projections? With further going evolution of the code, and approach with Ranges I guess they would be used on the <code>glue_<i>_whatever_</i>.h</code> pretty soon. Probably it's the right time to call pattern_sort with projection, no?

Reply:
In any case it cam be discussed later... Right now I don't see any problems on 1179 line and it can be merged as is.

Reply:
> I am trying to understand what is more correct design-wise

I considered introducing projections directly into the patterns like `__pattern_sort` below, but decided it is too early yet. I think we first need to do more prototyping for range support before making design decisions like this.